src/identitylookup.cs

```diff
@@ -0,0 +1,96 @@
+﻿//
+// IdentityLookup C# bindings
+//
+// Authors:
+//	Alex Soto  <alexsoto@microsoft.com>
+//
+// Copyright 2017 Xamarin Inc. All rights reserved.
+//
+
+#if XAMCORE_2_0
+using System;
+using XamCore.Foundation;
+using XamCore.ObjCRuntime;
+
+namespace XamCore.IdentityLookup {
+
+	[iOS (11,0)]
+	[Native]
+	public enum ILMessageFilterAction : nint {
+		None = 0,
+		Allow = 1,
+		Filter = 2,
+	}
+
+	[iOS (11,0)]
+	[ErrorDomain ("ILMessageFilterErrorDomain")]
+	[Native]
+	public enum ILMessageFilterError : nint {
+		System = 1,
+		InvalidNetworkUrl = 2,
+		NetworkUrlUnauthorized = 3,
+		NetworkRequestFailed = 4,
+		RedundantNetworkDeferral = 5,
+	}
+
+	[iOS (11,0)]
+	[DisableDefaultCtor]
+	[BaseType (typeof (NSObject))]
+	interface ILMessageFilterExtension {
+	}
+
+	[iOS (11,0)]
+	[DisableDefaultCtor]
+	[BaseType (typeof (NSExtensionContext))]
+	interface ILMessageFilterExtensionContext {
+
+		[Export ("deferQueryRequestToNetworkWithCompletion:")]
+		[Async]
+		void DeferQueryRequestToNetwork (Action<ILNetworkResponse, NSError> completion);
+	}
+
+	interface IILMessageFilterQueryHandling { }
+
+	[iOS (11,0)]
+	[Protocol]
+	interface ILMessageFilterQueryHandling {
+
+		[Abstract]
+		[Export ("handleQueryRequest:context:completion:")]
+		void HandleQueryRequest (ILMessageFilterQueryRequest queryRequest, ILMessageFilterExtensionContext context, Action<ILMessageFilterQueryResponse> completion);
```

Comment:
that's repeating `QueryRequest` from the first parameter 
but `Handle` is not very descriptive and already used for a property
can you just double-check the swift name ?